crates/sui-core/src/transaction_driver/effects_certifier.rs

```diff
@@ -248,7 +216,7 @@ impl EffectsCertifier {
             let name = *name;
             let raw_request = raw_request.clone();
             let future = async move {
-                // Keep retrying transient errors until cancellation.
+                // TODO(fastpath): add a timeout for the loop.
```

Comment:
how will this todo be different than the WAIT_FOR_EFFECTS_TIMEOUT below? You will have an even longer timeout that will abort the wait_for_acknowledgments call?

Reply:
The timeout on the loop will force retrying submission instead of individual WaitForEffects calls, which will return Expired eventually even if the txn has executed. But earlier retry of submission is probably not worth that much as an optimization. Dropping the TODO.